gc/base/MarkingScheme.hpp

```diff
@@ -169,43 +160,49 @@ class MM_MarkingScheme : public MM_BaseVirtual
 
 	void completeScan(MM_EnvironmentBase *env);
 	
-	MMINLINE void
-	scanObject(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, MM_CollectorLanguageInterface::MarkingSchemeScanReason reason)
+	/**
+	 * Public. Called from external context, eg concurrent GC.
+	 */
+	MMINLINE uintptr_t
+	scanObject(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, MM_MarkingDelegate::MarkingSchemeScanReason reason, uintptr_t sizeToDo = UDATA_MAX)
```

Comment:
See previous comment. At least, private scanObject() avoids unnecessary NULL check, and allows for more efficient updating of mark stats.